static-assets/components/cstudio-dialogs/uploadCMIS-dialog.js

```diff
@@ -0,0 +1,290 @@
+        var newdiv = YDom.get("cstudio-wcm-popup-div");
+        if (newdiv == undefined) {
+            newdiv = document.createElement("div");
+            document.body.appendChild(newdiv);
+        }
+
+        var divIdName = "cstudio-wcm-popup-div";
+        newdiv.setAttribute("id",divIdName);
+        newdiv.className= "yui-pe-content";
+        newdiv.innerHTML = '<div class="contentTypePopupInner" id="upload-popup-inner">' +
+            '<div class="contentTypePopupContent" id="contentTypePopupContent"> ' +
+            '<div class="contentTypePopupHeader">Upload</div> ' +
+            '<div><form id="asset_upload_form">' +
+            '<div class="contentTypeOuter">'+
+            '<div class="formDesc">Please select a file to upload</div> ' +
+            '<div><table><tr><td><input type="hidden" name="site_id" value="' + site + '"/></td>' +
+            '<td><input type="hidden" name="cmis_path" value="' + path + '"/></td></tr>' +
+            '<td><input type="hidden" name="cmis_repo_id" value="' + repositoryId + '"/></td></tr>' +
+            '<tr><td>File:</td><td><input type="file" name="file" id="uploadFileNameId"/></td></tr>' +
+            '</table></div>' +
+            '</div>' +
+            '<div class="contentTypePopupBtn"> ' +
+            '<input type="button" class="btn btn-primary cstudio-xform-button ok" id="uploadButton" value="Upload" disabled />' +
+            '<input type="button" class="btn btn-default cstudio-xform-button" id="uploadCancelButton" value="Cancel"  /></div>' +
+            '</form></div>' +
+            '<div><div  style="visibility:hidden; margin-bottom:1.5em;" id="indicator">Uploading...</div>' +
+            '</div> ' +
+            '</div>';
+
+        document.getElementById("upload-popup-inner").style.width = "350px";
+        document.getElementById("upload-popup-inner").style.height = "180px";
+
+        // Instantiate the Dialog
+        upload_dialog = new YAHOO.widget.Dialog("cstudio-wcm-popup-div",
+            { width : "360px",
+                height : "242px",
+                effect:{
+                    effect: YAHOO.widget.ContainerEffect.FADE,
+                    duration: 0.25
+                },
+                fixedcenter : true,
+                visible : false,
+                modal:true,
+                close:false,
+                constraintoviewport : true,
+                underlay:"none"
+            });
+
+        // Render the Dialog
+        upload_dialog.render();
+
+        var filenameInput = document.getElementById("uploadFileNameId");
+        YAHOO.util.Event.addListener(filenameInput, "change", this.uploadFileEvent);
+
+        var eventParams = {
+            self: this
+        };
+
+        YAHOO.util.Event.addListener("uploadButton", "click", this.uploadPopupSubmit, eventParams);
+        YAHOO.util.Event.addListener("uploadCancelButton", "click", this.uploadPopupCancel);
+
+        $("body").on("keyup", "#cstudio-wcm-popup-div", function(e) {
```

Comment:
This will persist past the dialog getting closed if not closed using the escape key. You should detach this event when the dialog is disposed from any means.

Reply:
Done